File: binary_search.py
def binarySearch(list, item):
    low = 0
    high = len(list) - 1
    while low <= high:
        mid = (high + low) // 2
        guess = list[mid]
        if guess == item:
            return mid
        if guess > item:
            high = mid - 1
        else:
            low = mid + 1
    return None


ls = [1, 3, 4, 5, 6, 8, 10, 12]
print(binarySearch(ls, 8))

File: binary_search.js
function binarySearch(list, item) {
  var low = 0;
  var high = list.length - 1;
  while (low <= high) {
    var mid = Math.floor((high - low) / 2);
    var guess = list[mid];
    if (guess === item) {
      return mid;
    }
    if (guess < item) {
      low = mid + 1;
    } else if (guess > item) {
      high = guess - 1;
    }
  }
  return null;
}

var ls = [1, 3, 4, 5, 6, 8, 10, 12];
console.log(binarySearch(ls, 3));
